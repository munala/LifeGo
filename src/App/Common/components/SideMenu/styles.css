.side-menu-container {
  min-width:220px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: stretch;
  padding-right: 24px;
  font-size: 14px;
  font-family: Roboto,RobotoDraft,Helvetica,Arial,sans-serif;
}

.side-menu-items {
  padding-right: 24px;
  width:100%;
  display: flex;
  height: 100%;
  flex-direction: column;
  justify-content: flex-start;
  align-items: stretch;
}

.side-menu-item {
  text-decoration: none;
  padding-left: 24px;
  display: flex;
  height: 48px;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  cursor: pointer;
  color: rgba(0,0,0,0.87);
}

.side-menu-item:hover {
  background-color: #e7e7e7;
}

.footer {
  padding-left: 24px;
  margin-right: 24px;
  display: flex;
  color: #BCBCBC;
  margin-bottom: 24px;
  font-size: 12px;
  font-weight: lighter;
}

.divider {
  margin-top: 10px;
  margin-bottom: 10px;
  display: flex;
  height: 1px;
  background-color: #ddd;
}

.item-title {
  display: flex;
  padding-left: 24px;
  color: #3F3F3F;
}

.item-title-bottom {
  display: flex;
  color: #828282;
}
