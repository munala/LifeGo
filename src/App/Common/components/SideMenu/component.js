import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Icon from '@material-ui/core/Icon';

import './styles.css';

const SideMenu = ({
  menuItems: { top, bottom },
  activeItem,
  location: { pathname },
}) => {
  const path = pathname === '/home' ? '/' : pathname;
  return (
    <div className="side-menu-container" >
      <div className="side-menu-items">
        {top.map(item => (
          <Link className="side-menu-item" key={item.icon} to={item.to}>
            <Icon
              style={{ color: (item.to.pathname || item.to) === path ? '#00bcd4' : '#888' }}
            >
              {item.icon}
            </Icon>

            <div
              className="item-title"
              style={{ color: (item.to.pathname || item.to) === path ? '#00bcd4' : '#3F3F3F' }}
            >
              {item.text}
            </div>
          </Link>
      ))}

        <div className="divider" />

        {bottom.map(item => (
          <Link className="side-menu-item" key={item.text} to={item.to}>
            <div
              className="item-title-bottom"
              style={{ color: item.to === path ? '#00bcd4' : '#3F3F3F' }}
            >
              {item.text}
            </div>
          </Link>
      ))}
      </div>

      <div className="footer">Â© 2018 Oliver Munala</div>
    </div>
  );
};
SideMenu.propTypes = {
  menuItems: PropTypes.shape({
    top: PropTypes.arrayOf(PropTypes.shape({})),
    bottom: PropTypes.arrayOf(PropTypes.shape({})),
  }).isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
  activeItem: PropTypes.string.isRequired,
};

export default SideMenu;
