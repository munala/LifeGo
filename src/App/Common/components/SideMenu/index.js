import React from 'react';
import PropTypes from 'prop-types';

import Icon from '@material-ui/core/Icon';
import './styles.css';

const Header = ({
  menuItems: { top, bottom },
  activeItem,
}) => (
  <div className="side-menu-container" >
    <div className="side-menu-items">
      {top.map(item => (
        <div className="side-menu-item" key={item.icon}>
          <Icon
            style={{ color: item.icon === activeItem ? '#00bcd4' : '#888' }}
          >
            {item.icon}
          </Icon>

          <div
            className="item-title"
            style={{ color: item.icon === activeItem ? '#00bcd4' : '#3F3F3F' }}
          >
            {item.text}
          </div>
        </div>
      ))}

      <div className="divider" />

      {bottom.map(item => (
        <div className="side-menu-item" key={item.text}>
          <div className="item-title-bottom" >{item.text}</div>
        </div>
      ))}
    </div>

    <div className="footer">Footer</div>
  </div>
);

Header.propTypes = {
  menuItems: PropTypes.shape({
    top: PropTypes.arrayOf(PropTypes.shape({})),
    bottom: PropTypes.arrayOf(PropTypes.shape({})),
  }).isRequired,
  activeItem: PropTypes.string.isRequired,
};

export default Header;
