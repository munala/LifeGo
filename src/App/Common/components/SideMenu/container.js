import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';
import { logout } from '../../../Auth/actions';

import SideMenu from './component';

const mapStateToProps = ({
  loggedIn,
  profile,
  ...state
}) => {
  const loggedOutMenuItems = {
    top: [
      {
        to: '/',
        icon: 'explore',
        text: 'Explore',
      },
      {
        to: '/login',
        icon: 'add',
        text: 'Join LifeGo',
      },

    ],
    bottom: [],
  };

  const loggedInMenuItems = {
    top: [
      {
        to: '/',
        icon: 'home',
        text: 'Home',
      },
      {
        to: '/explore',
        icon: 'explore',
        text: 'Explore',
      },
      {
        to: '/profile',
        icon: 'account_circle',
        text: 'Profile',
      },
    ],
    bottom: [
      {
        to: '/settings',
        text: 'Settings',
      },
    ],
  };

  const menuItems = loggedIn ? loggedInMenuItems : loggedOutMenuItems;

  return ({
    menuItems,
  });
};

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({
    logout,
  }, dispatch),
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SideMenu));
