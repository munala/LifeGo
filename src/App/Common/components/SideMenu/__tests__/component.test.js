import React from 'react';
import { shallow } from 'enzyme';
import SideMenu from '../container';

describe('SideMenu tests', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<SideMenu
      menuItems={{ top: [], bottom: [] }}
      activeItem="home"
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
