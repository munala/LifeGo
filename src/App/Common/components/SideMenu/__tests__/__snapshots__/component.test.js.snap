// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu tests renders without crashing 1`] = `<Route />`;
